provider/app_test.go

```diff
@@ -246,4 +248,75 @@ func TestApp(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("Hidden", func(t *testing.T) {
+		t.Parallel()
+
+		cases := []struct {
+			name   string
+			config string
+			hidden bool
+		}{{
+			name: "Is Hidden",
+			config: `
+			provider "coder" {}
+			resource "coder_agent" "dev" {
+				os = "linux"
+				arch = "amd64"
+			}
+			resource "coder_app" "test" {
+				agent_id = coder_agent.dev.id
+				slug = "test"
+				display_name = "Testing"
+				url = "https://google.com"
+				external = true
+				hidden = true
```

Comment:
@DanielleMaywood @johnstcn  if the app is `hidden`, does it make sense to define `display_name`, `url`, etc.? I don't know the answer, thinking loud here. If so:
* we could improve the validation to check conflicting values of properties
* we should check if coder/coder does not break as it could be expecting all of them to be defined

Reply:
I'd say dropping a `diag.Warn` is the most we should do here.
For the latter, I believe Danielle already tested manually, but I suppose we could add an integration test that defines both with and without.